src/components/TimePicker.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import styled from 'styled-components'
+import { BaseDateTimePickerProps, DateTimePicker } from '@material-ui/pickers'
+import { TextField } from '@material-ui/core'
+import DateFnsAdapter from '@material-ui/pickers/adapter/date-fns'
+import { Control, UseFormMethods, FieldError, Controller } from 'react-hook-form'
+
+import { TradeFormData } from './TradeWidget'
+import { FormInputError } from './TradeWidget/FormMessage'
+
+interface DateTimePickerControlProps<T> extends BaseDateTimePickerProps {
+  control: Control<T>
+  onClose?: () => void
+  customOnChange: (date: Date, keyboardInputValue?: string | undefined) => void
+  formValues: {
+    value: string | null
+    setValue: UseFormMethods['setValue']
+    errors?: FieldError
+    inputName: keyof T
+  }
+}
+
+const DateTimePickerControl: React.FC<DateTimePickerControlProps<TradeFormData>> = ({
+  control,
+  formValues,
+  minDate = Date.now(),
+  customOnChange,
+  ...restProps
+}) => {
+  const memoizedDateAdapter = React.useMemo(() => {
+    return new DateFnsAdapter()
+  }, [])
```

Comment:
Does this need to be inside component?

Reply:
according to docs, yes, but i think this only applies when switching timezones so :+1: